greymatter-io/operator
# Fixing state backup

[sc-27113]

## How to test

1. Build and deploy the operator from this branch.
2. Once the mesh is stable, look for log messages in the operator logs with `Waiting another 30 seconds for Redis availability` and then eventually `Connected to Redis for state backup`. The waiting message is back with this fix, indicating the state backup loop is actually running again.
3. Shell into the greymatter-datastore (redis) pod's "redis" container and run `keys *` to confirm the presence of the `gm-operator.gmHashes` and `gm-operator.k8sHashes` keys. They weren't being saved before this fix.
4. Bounce the operator and watch for these lines:
```
{"level":"info","ts":1664491114.3849797,"logger":"gitops","msg":"Connected to Redis for state backup"}
{"level":"info","ts":1664491114.38629,"logger":"gitops","msg":"Successfully loaded GM object hashes from Redis","key":"gm-operator.gmHashes"}
{"level":"info","ts":1664491114.3872986,"logger":"gitops","msg":"Successfully loaded K8s object hashes from Redis","key":"gm-operator.k8sHashes"}
```
